Fake telemetry

jamesridley opened the pull request:
## Description
Implement fake telemetry send and receive loops (using flag --fake_telemetry). As soon as state machine enters a state where it expects a new message from mission control, the fake receive loop will automatically "receive" the message to transition to the next state. For example: as soon as state machine enters the idle state, the fake receive loop will receive a fake calibrate message.

## Changes
- Added fake-recvloop 
- Added fake-sendloop so that there isn't an error when the pod tries to send a message to through a non-existent socket

## Additional Info 
Use --fake_telemetry to run. TODO: re-implement the fake system using an interface instead of the old flags system.

mkristien commented:
I wanted to discuss this last Saturday but I will just state my comments/suggestions here.

For the configuration, I would like to move stuff to the `utils/config.hpp`. Can you move the fake flags there?
Furthermore, the every thing that can be runtime configured should also be supported by using the configurations file. Telemetry already has a subfile `configurations/telemetry.txt` you can use for this. Alternatively, create a `configurations/fake_telemetry.txt` to encapsulate any fake setup in one place and not mix it with the real setup.

For the telemetry interface, it would be great if you managed to use the new interface infrastructure, starting with adding the sender interface to the `INTERFACE_LIST` in `utils/interfaces.hpp`. You can also look at examples of usage in `demo/implementation[1,2].cpp`. A brief guide is coming shortly.

iainmacpherson commented:
I've added a guide on this to the wiki. It can be found [here](https://github.com/Hyp-ed/hyped-2020/wiki/Using-the-new-simulation-config-system).

jamesridley commented:
> For the telemetry interface, it would be great if you managed to use the new interface infrastructure, starting with adding the sender interface to the `INTERFACE_LIST` in `utils/interfaces.hpp`. You can also look at examples of usage in `demo/implementation[1,2].cpp`. A brief guide is coming shortly.

Ok, I've re-implemented the fake system by faking client instead of the two loops, using a client interface that both fake and real client inherit from. I've removed the old fake loops and added the client interface to the list and it seems to work great when testing on a laptop.

iainmacpherson commented on src/telemetry/client.hpp:
```diff
@@ -25,26 +25,29 @@
+class Client : public ClientInterface {
   public:
-    explicit Client(Logger& log);
+    explicit Client();
```
```suggestion
    Client();
```

iainmacpherson replied:
You don't need to use the explicit keyword here.

```suggestion
    Client();
```
